app/features/conference/components/Conference.js

```diff
@@ -131,7 +149,115 @@ class Conference extends Component<Props, *> {
         setupWiFiStats(iframe);
 
         this._api.on('readyToClose', () => this._navigateToHome());
+
+        this._setAvatarURL(this.props._avatarURL);
+        this._setEmail(this.props._email);
+        this._setName(this.props._name);
+
+        this._api.on('videoConferenceJoined',
+            (conferenceInfo: Object) =>
+                this._onVideoConferenceJoined(conferenceInfo));
+    }
+
+    /**
+     * Updates redux state's user name from conference.
+     *
+     * @param {Object} params - Returned object from event.
+     * @param {string} id - Local Participant ID.
+     * @returns {void}
+     */
+    _onDisplayNameChange(params: Object, id: string) {
```

Comment:
This should also trigger an update on the avatar.

Reply:
Agreed.
I'll cover this with middleware thing.